ksim/losses/loss.py

```diff
@@ -0,0 +1,13 @@
+"""Defines the base loss class, along with some implementations."""
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+import equinox as eqx
+from jaxtyping import Array
+
+
+class Loss(ABC):
+    @abstractmethod
+    def compute_loss(self, agent: eqx.Module, batch: Any, output: Any) -> Array:  # noqa: ANN401
+        """Compute the loss for a given model and batch."""
```

Comment:
i don't think it really makes sense to have this as our base abstract class - losses don't really follow this interface (e.g., the ppo loss below has a completely different function signature)

better to just have some sort of utility class exclusively for ppo loss, don't need to have this here